server/src/main/java/org/candlepin/policy/criteria/CriteriaRules.java

```diff
@@ -65,7 +65,7 @@ public CriteriaRules(Configuration config, ConsumerCurator consumerCurator) {
         // consumer UUID
         Consumer hostConsumer = null;
         if (consumer.getFact("virt.uuid") != null) {
-            hostConsumer = consumerCurator.getHost(consumer.getFact("virt.uuid"), consumer.getOwner());
+            hostConsumer = consumerCurator.getHost(consumer);
```

Comment:
@vritant @awood It is worth noting that my query tests were done directly with SQL on the DB. We need to make sure that hibernate isn't performing a query any more complex than selecting the GuestId by GuestId.guest_id

My test query (see the design doc for how many records etc...
```sql
select guest_id, uuid, o.account from cp_consumer_guests as g join cp_consumer as h on h.id=g.consumer_id join cp_owner as o on h.owner_id=o.id where g.guest_id='4230b451-7bda-010d-be10-cacf20aff6a7';
```